[hanscau] iP

hanscau opened the pull request:
### Snah

> "The only chatbot you need, is the one you find along the way." - Someone Probably ([source](https://en.wikipedia.org/wiki/Probably))

The ultimate tasks manager, only the best for the best. Features includes:
- CLI based
- Nothing to learn
- Local save
- ~VERY~ _EXTREMELY_ good

How to setup
1. Pull this repo
2. Run `java snah.java`
3. Feel your tasks melt into nothingness 🤤 

And it is **ABSOLUTELY FREE** (for now...)

Features:

- [x] Manage tasks
- [ ] Do your homework
- [ ] LLM integrated, Blockchain connected, AI capable, Cloud based

---

If you know how to read code. Here my letter from `me` to you:
```python
if reader == 'code reader':
    print("Hello :)")
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

hanscau committed:
Rename chatbot

hanscau committed:
Add format and echo

hanscau committed:
Add list function

hanscau committed:
Add tasks

hanscau committed:
Add deadline, event and task

hanscau committed:
Add testcases

hanscau committed:
Add error handling

hanscau committed:
Add testcase

hanscau committed:
Refactor to ArrayList

hanscau committed:
Add delete feature

hanscau committed:
Add testcases

hanscau committed:
Add save feature

hanscau committed:
Add ignore save files

hanscau committed:
Merge branch 'branch-Level-7'

hanscau committed:
Add java date time for deadline

hanscau committed:
Add clear function

hanscau committed:
Fix test cases

hanscau committed:
Merge remote-tracking branch 'origin/branch-Level-8'

hanscau committed:
Refactor storage

hanscau committed:
Remove useless imports

hanscau committed:
Refactor parser

hanscau committed:
Refactor ui

hanscau committed:
Refactor tasklist

hanscau committed:
Merge branch 'A-MoreOOP'

hanscau committed:
Add everything into packages

hanscau committed:
Merge branch 'branch-A-Packages'

hanscau committed:
Add javadocs

hanscau committed:
Remove unused import

hanscau committed:
Add find function

hanscau committed:
Merge branch 'branch-A-JavaDoc'

hanscau committed:
Merge branch 'branch-Level-9'

hanscau committed:
Merge branch 'branch-A-CodingStandard'

hanscau committed:
Merge remote-tracking branch 'origin/add-gradle-support'

hanscau committed:
Edit gradle settings

hanscau committed:
Fix import

hanscau committed:
Merge commit '9670b22cc419f0651c16b7f4d8bb875ad0c015d5'

hanscau committed:
Add JUnit tests

hanscau committed:
Merge branch 'branch-A-JUnit'

AlekKwek18 commented:
Overall LGTM, clear and concise javadocs. Mostly followed good coding standard.

AlekKwek18 commented on src/main/java/snah/Tasklist.java:
```diff
@@ -0,0 +1,54 @@
+import snah.task.Task;
+import snah.util.Storage;
+
+public class Tasklist {
```
Good to include javadocs for Tasklist methods!

AlekKwek18 replied:
one suggestion is to change the class name from Tasklist to TaskList as class names should follow the PascalCase convention****

AlekKwek18 commented on src/main/java/snah/Snah.java:
```diff
@@ -0,0 +1,190 @@
+public class Snah {
+
+    private Storage storage;
+    private Tasklist tasksList;
```
Good use of camelCase for variables

jewiteo commented on src/main/java/snah/task/Task.java:
```diff
@@ -0,0 +1,59 @@
+     * @return Task in the format to save to file
+     */
+    public String toSaveFile() {
+        return String.format("T:%s:%s", isDone() ? "x" : "", getDescription());
```
Good use of ternary operator to keep code concise :)

jewiteo commented on src/main/java/snah/util/Parser.java:
```diff
@@ -0,0 +1,146 @@
+     */
+    public static Command getCommand(String input) {
+        String commandSymbol = input.split(" ", 2)[0].toUpperCase();
+        switch (commandSymbol) {
```
Case statement should not have indentations

jewiteo commented on src/test/java/snah/util/ParserTest.java:
```diff
@@ -0,0 +1,60 @@
+
+public class ParserTest {
+    @Test
+    public void getEventPayloadTest() {
```
Good use of logical blocks for the different test cases but maybe you could separate out the different conditions into different test and name it according to the three part format

jewiteo reviewed (commented):
Overall code is concise and easy to understand with minimal areas to improve on :)))

hanscau committed:
Add checkstyle config

hanscau committed:
Add checkstyle extension

hanscau committed:
Fix all checkstyle issues

hanscau committed:
Merge remote-tracking branch 'origin/branch-A-CheckStyle'

hanscau committed:
Add JavaFX

hanscau committed:
Refactor to FXML

hanscau committed:
Merge remote-tracking branch 'origin/branch-Level-10'

hanscau committed:
Add assert for switch case default

The default switch case for getResponse should never be accessed because all invalid commands
will be parsed into the enum INVALID.

hanscau committed:
Refactor duplicated code

Previously duplicated code is removed to make sure the code base is maintainable. With
the inclusion of JavaFX into the code base, the chatLoop can be done using the getResponse method instead.

hanscau committed:
Add assert to getTaskLists with invalid task type

Since the save file is managed by the program itself, there should never be
an instance where there is a task type that is not recognised by the program
itself. If such a case occur, it means that the save file have
been tampered with and data integrity has been breached.

hanscau committed:
Merge pull request #2 from hanscau/branch-A-Assertions

Add assertion to program

hanscau committed:
Refactor wrong task message

hanscau committed:
Refactor boolean to a better name

hanscau committed:
Add better comments

hanscau committed:
Refactor commands to individual classes

To prevent long switch case statements of how to handle each commands and instead just have the command create a command class
that handle how each items are being handled, it is better for error handling too

Note:
Each commands are not placed inside a hashmap to ensure that listing all commands will lists all available commands

hanscau committed:
Merge commit '10b4c5ca4e3f691bba75516130da26efeb2735bd' into branch-A-CodeQuality

hanscau committed:
Fix exiting in CLI mode

hanscau committed:
Add handle invalid tasks index

hanscau committed:
Refactor error message

hanscau committed:
Merge pull request #3 from hanscau/branch-A-CodeQuality

Better Code Quality

hanscau committed:
Merge commit 'edca0b91f379084787f0e683c6077859eb93afd4' into branch-A-Streams

hanscau committed:
Refactor arrays to streams

hanscau committed:
Merge pull request #4 from hanscau/branch-A-Streams

Add Stream

hanscau committed:
Fix Parser test case

hanscau committed:
Add help command

hanscau committed:
Add tutorial tasks for new user

hanscau committed:
Merge branch 'branch-C-Help'

hanscau committed:
refactor save file delimiter to ;

hanscau committed:
Add more error handling

hanscau committed:
Merge branch 'branch-A-MoreErrorHandling'

hanscau committed:
Add exit command

hanscau committed:
Add app title

hanscau committed:
Fix better tutorial task

hanscau committed:
Edit Readme for the application
